execution/executor-benchmark/src/block_partitioning.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © Aptos Foundation
+
+use crate::pipeline::ParToExeMsg;
+use aptos_block_partitioner::sharded_block_partitioner::ShardedBlockPartitioner;
+use aptos_crypto::HashValue;
+use aptos_logger::info;
+use aptos_types::{
+    block_executor::partitioner::{
+        CrossShardDependencies, ExecutableBlock, ExecutableTransactions,
+        TransactionWithDependencies,
+    },
+    transaction::Transaction,
+};
+use std::{
+    sync::mpsc::Receiver,
+    time::Instant,
+};
+use std::sync::mpsc::SyncSender;
+
+pub(crate) struct BlockPartitioningStage {
+    num_iterations: usize,
+    executable_block_sender: SyncSender<ParToExeMsg>,
+    maybe_exe_fin_receiver: Option<Receiver<()>>,
+    maybe_partitioner: Option<ShardedBlockPartitioner>,
+}
+
+impl BlockPartitioningStage {
+    pub fn new(
+        executable_block_sender: SyncSender<ParToExeMsg>,
+        maybe_exe_fin_receiver: Option<Receiver<()>>,
+        num_shards: usize,
+    ) -> Self {
+        let maybe_partitioner = if num_shards <= 1 {
+            None
+        } else {
+            let partitioner = ShardedBlockPartitioner::new(num_shards);
+            Some(partitioner)
+        };
+
+        Self {
+            num_iterations: 0,
+            executable_block_sender,
+            maybe_exe_fin_receiver,
+            maybe_partitioner,
+        }
+    }
+
+    pub fn process(&mut self, mut txns: Vec<Transaction>) {
+        let current_block_start_time = Instant::now();
+        info!(
+            "In iteration {}, received {:?} transactions.",
+            self.num_iterations,
+            txns.len()
+        );
+        let block_id = HashValue::random();
+        let block: ExecutableBlock<Transaction> = match &self.maybe_partitioner {
+            None => (block_id, txns).into(),
+            Some(partitioner) => {
+                let last_txn = txns.pop().unwrap();
+                assert!(matches!(last_txn, Transaction::StateCheckpoint(_)));
+                let analyzed_transactions = txns.into_iter().map(|t| t.into()).collect();
```

Comment:
Hmm.. may be we need to do this in parallel otherwise parsing txn to convert to analyzed txn is going to be to slow..

Reply:
this is far from being the bottleneck. Fine for now?